File: javascript/random_2/random.js
const generateRandomNumber = (from, to) => {
    return Math.floor(Math.random() * (to - from) + from)
}

const first = generateRandomNumber(5, 231)
const second = generateRandomNumber(5, 231)
const third = generateRandomNumber(5, 231)

const findBiggestNumber = (first, second, third) => {
    return Math.max(first, second, third)
}

const findSmallestNumber = (first, second, third) => {
    return Math.min(first, second, third)
}

const findMiddleNumber = (first, second, third) => {
    return (first + second + third) - findBiggestNumber(first, second, third) - findSmallestNumber(first, second, third)
}

const calculateAverage = (first, second, third) => {
    const middleNumber = findMiddleNumber(first, second, third)
    const smallest = findSmallestNumber(first, second, third).toString()
    const biggest = findBiggestNumber(first, second, third).toString()
    
    return (middleNumber
        + parseInt(smallest.split('')[smallest.length - 1])
        + parseInt(biggest.split('')[biggest.length - 1])) / 2
}

console.log(`Wylosowano liczby ${first}, ${second}, ${third}.`)

console.log(`Największa z wylosowanych liczb to: ${findBiggestNumber(first, second, third)}`)
console.log(`Najmniejsza z wylosowanych liczb to: ${findSmallestNumber(first, second, third)}`)
console.log(`Średnia wynosi: ${calculateAverage(first, second, third)}`)

File: python/random_2/random.py
import random


def generate_number(minimum, maximum):
    return random.randint(minimum, maximum)


number_1 = generate_number(5, 231)
number_2 = generate_number(5, 231)
number_3 = generate_number(5, 231)


def get_numbers_in_order(num_1, num_2, num_3):
    min_value = min(num_1, num_2, num_3)
    max_value = max(num_1, num_2, num_3)
    mid_value = num_1 + num_2 + num_3 - min_value - max_value
    return min_value, mid_value, max_value


(min_v, mid_v, max_v) = get_numbers_in_order(number_1, number_2, number_3)


def get_average(num_1, num_2, num_3):
    all_numbers = get_numbers_in_order(num_1, num_2, num_3)
    return (all_numbers[1] + int(str(all_numbers[0])[-1]) + int(str(all_numbers[2])[-1])) / 2


print(f'Największa z wylosowanych liczb to: {get_numbers_in_order(number_1, number_2, number_3)[2]}')
print(f'Najmniejsza z wylosowanych liczb to: {get_numbers_in_order(number_1, number_2, number_3)[0]}')
print(f'Średnia wynosi: {get_average(number_1, number_2, number_3)}')
